cfgrammar/src/lib/u32struct.rs

```diff
@@ -71,6 +76,53 @@ macro_rules! u32struct {
                 u32::from(st.v)
             }
         }
+
+        impl TryFrom<$n> for u8 where usize: From<$n>
+        {
+            type Error = TryFromIntError;
+
+            fn try_from(st: $n) -> Result<Self, Self::Error> {
+                Ok(u8::try_from(usize::from(st.v))?)
+            }
+        }
+    }
+}
+
+macro_rules! transitionmacro {
+    ($(#[$attr:meta])* $n: ident) => {
+        $(#[$attr])*
+        #[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+        #[cfg_attr(feature="serde", derive(Serialize, Deserialize))]
+        pub struct $n<T>(pub T);
+
+        impl<T: PrimInt + Unsigned> From<usize> for $n<T> {
+            fn from(v: usize) -> Self {
+                $n(num_traits::cast(v).unwrap())
+            }
+        }
+
+        impl<T: PrimInt + Unsigned> From<$n<T>> for usize {
+            fn from(st: $n<T>) -> Self {
+                num_traits::cast(st.0).unwrap()
+            }
+        }
+
+        impl<T: PrimInt + Unsigned> From<$n<T>> for u32 where T: Unsigned {
+            fn from(st: $n<T>) -> Self {
+                num_traits::cast(st.0).unwrap()
+            }
+        }
+
+        impl<T: PrimInt + Unsigned> TryFrom<$n<T>>
+                                for u8
+                              where T: Unsigned, usize: From<T>
```

Comment:
Should the `for` and the `where` be aligned here?

Reply:
Oops, somehow I pressed return randomly here (dunno how). Fixed.